### PR TITLE
Impl 214 and wip 215

### LINKED ISSUE #214: UI - runs filters extraction and use them in workflows datagrid
UI - runs filters should be extracted outside of runs folder that they can be used in workflows data grid as well.

Screenshot 2020-06-22 at 15 06 18

### LINKED ISSUE #215: UI - Workflows datagrid filters/sorts
UI - Workflows datagrid filters/sorts. (Should be implemented same way as it is in Runs datagrid)
+ reset buttons.

blocked by: #214 

Screenshot 2020-06-22 at 15 07 19

### PR DESCRIPTION
**Pull request by senelesithole**
fixes #214 and 
WIP #215 

Requires input on.
1. New button on Workflows home, aligned with 'create workflow' button or okay as is.
2. Clear sort functionality.

### CODE REVIEW

File: ui/src/app/components/workflows/workflows-home/workflows-home.component.html [hunk 1]
```diff
@@ -24,8 +24,37 @@
...
+      </button>
+      <button type="button" class="btn btn-sm btn-secondary" (click)="clearSort()">
+        <clr-icon shape="trash"></clr-icon>
+        Clear sort
+      </button>
+    </div>
+  </clr-dg-action-bar>
+
+  <clr-dg-column [clrDgField]="workflowsHomeColumns.WORKFLOW_NAME" [clrFilterValue]>
+    <ng-container *clrDgHideableColumn="{hideden: false}">Workflow name</ng-container>
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflows-home/workflows-home.component.html]: typo: hidden

File: ui/src/app/components/workflows/workflows-home/workflows-home.component.html [hunk 2]
```diff
@@ -24,8 +24,37 @@
...
+    <clr-dg-filter [clrDgFilter]="workflowNameFilter">
+      <app-string-filter #workflowNameFilter [property]="workflowsHomeColumns.WORKFLOW_NAME"
+        [removeFiltersSubject]="removeWorkflowFilterSubject">
+      </app-string-filter>
+    </clr-dg-filter>
+  </clr-dg-column>
+
+  <clr-dg-column [clrDgField]="workflowsHomeColumns.PROJECT_NAME" [clrDgSortOrder]="ascSort" [(clrDgSorted)]="sorted">
+    <ng-container *clrDgHideableColumn="{hideden: false}">Project name</ng-container>
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflows-home/workflows-home.component.html]: typo hidden

File: ui/src/app/components/workflows/workflows-home/workflows-home.component.html [hunk 3]
```diff
@@ -24,8 +24,37 @@
...
+    <clr-dg-filter [clrDgFilter]="projectNameFilterHome">
+      <app-string-filter #projectNameFilterHome [property]="workflowsHomeColumns.PROJECT_NAME"
+        [removeFiltersSubject]="removeWorkflowFilterSubject">
+      </app-string-filter>
+    </clr-dg-filter>
+  </clr-dg-column>
+
   <clr-dg-column>Is active</clr-dg-column>
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflows-home/workflows-home.component.html:58]: No filter for is active?

**Review by kevinwallimann — COMMENTED** [review]
If I delete /run a workflow, the filters are reset. I think a better approach would be to retain the filters, I guess this could be achieved with onClarityDgRefresh, similarly as in runs.component

File: ui/src/app/components/common/datagrid/filters/datetime-range-filter/datetime-range-filter.component.ts [hunk 4]
```diff
@@ -15,10 +15,10 @@
 
 import { AfterViewInit, Component, Input } from '@angular/core';
 import { ClrDatagridFilterInterface } from '@clr/angular';
-import { DagRunModel } from '../../../../models/dagRuns/dagRun.model';
+import { DagRunModel } from '../../../../../models/dagRuns/dagRun.model';
```
Comment (kevinwallimann) [inline ui/src/app/components/common/datagrid/filters/datetime-range-filter/datetime-range-filter.component.ts:18]: Interesting that the filters work, even though you're not passing a `DagRunModel`. I'm thinking, if we generalize the filters, then probably the filters should not implement
`ClrDatagridFilterInterface<DagRunModel>`, but `ClrDatagridFilterInterface<any>` instead. The type parameter is only used for the `accepts` method, and there we have a type-unsafe access anyway with `item[this.property]`, i.e. we can't make this component type-safe anyway, so we don't lose anything by replacing `DagRunModel` with `any`

File: ui/src/app/services/workflow/workflow.service.ts [hunk 5]
```diff
@@ -40,6 +42,18 @@ export class WorkflowService {
       .pipe(map((_) => _.body));
   }
 
+  searchProjects(searchRequestModel: TableSearchRequestModel): Observable<TableSearchResponseModel<ProjectModel>> {
```
Comment (jozefbakus) [inline ui/src/app/services/workflow/workflow.service.ts]: You dont need this because we will sort and filter on FE

**Review by jozefbakus — CHANGES_REQUESTED** [review]
1. Move filters outside of runs - DONE
2. You dont have to call backend search endpoint for workflows filter/sort
5. You have to persist filters/sorts to store. and use them when page is reloaded 
3. How to pre set sort https://clarity.design/documentation/datagrid/custom-sorting
4. How to pre set filter https://clarity.design/documentation/datagrid/built-in-filters (end of the doc)

**Review by jozefbakus — COMMENTED** [review]
could you please remove logs/app.log from changes.

**Pull request closed without merging**